src/instruments/src/EFB/Utils/AircraftVersionChecker.ts

```diff
@@ -0,0 +1,288 @@
+                    version: json.version,
+                });
+            });
+        });
+        return this.buildInfo;
+    }
+
+    /**
+     * Parses the version string and returns the version info as VersionInfoData object.
+     * Note: public because of jest test
+     *
+     * @param versionString as provided by the a32nx_build_info.json file.
+     * @throws Error if the version string is not in the correct format.
+     */
+    public static getVersionInfo(versionString: string): VersionInfoData {
+        const matchBuildInfo = versionString.match(/^v?((\d+)\.(\d+)\.(\d+))-(.*)\.(.{7})$/);
+        if (matchBuildInfo) {
+            const version = matchBuildInfo[1];
+            const major = parseInt(matchBuildInfo[2], 10);
+            const minor = parseInt(matchBuildInfo[3], 10);
+            const patch = parseInt(matchBuildInfo[4], 10);
+            const branch = matchBuildInfo[5];
+            const commit = matchBuildInfo[6];
+            return {
+                version,
+                major,
+                minor,
+                patch,
+                branch,
+                commit,
+            };
+        }
+        throw new Error('Invalid version format');
+    }
+
+    /**
+     * Retrieves the various versions from the current aircraft and GitHub and stores them in class variables.
+     *
+     * @private
+     */
+    private static async initialize() {
+        this.releaseInfo = await GitVersions.getReleases('flybywiresim', 'a32nx', false, 0, 1);
+        this.newestCommit = await GitVersions.getNewestCommit('flybywiresim', 'a32nx', 'master');
+        this.newestExpCommit = await GitVersions.getNewestCommit('flybywiresim', 'a32nx', 'experimental');
+        this.buildInfo = await AircraftVersionChecker.getBuildInfo();
+    }
+
+    /**
+     * Checks if the given version is outdated and shows a notification if it is.
+     *
+     * @param versionInfo
+     * @returns true if the version is outdated, false otherwise.
+     * @private
+     */
+    private static checkOutdated(versionInfo: VersionInfoData): boolean {
+        // Set branchName to the long versions of the aircraft edition names
+        const branchName = this.getBranchName(versionInfo);
+
+        // Check if main version is outdated
+        if (Compare(versionInfo.version, this.releaseInfo[0].name) < 0) {
+            console.log(`New version available: ${versionInfo.version} ==> ${this.releaseInfo[0].name}`);
+            this.showVersionPopup('', versionInfo.version, this.releaseInfo[0].name);
+            return true;
+        }
+
+        // If the user's version is equal or newer than the latest release then check if
+        // the edition is Development or Experimental and if the commit is older than
+        // {maxAge} days after the latest release to show notification
+
+        const maxAge = 3;
+        const timestampAircraft: Date = new Date(this.buildInfo.built);
+
+        if ((branchName === 'Development')
```

Comment:
I would store those branch names in an enum `KnownBranchNames` to avoid having magic values in the code

Reply:
Good idea - done